soc/nxp/lpc/lpc55xxx/Kconfig.defconfig

```diff
@@ -36,17 +36,14 @@ config LPC55XXX_SRAM_CLOCKS
 # By default, USB RAM is assumed to be present.
 # Disable this Kconfig in case there is no dedicated USB RAM.
 config LPC55XXX_USB_RAM
-	default y
+	default y if (!SOC_LPC55S06 && !SOC_LPC55S36)
 
 # Set to the minimal size of data which can be written.
 config FLASH_FILL_BUFFER_SIZE
 	default 512
 
 if SOC_LPC55S06
 
-config LPC55XXX_USB_RAM
-	default n
```

Comment:
so was this override not occuring?

Reply:
No, it was not.

Reply:
(scratching head)

Reply:
> No, it was not.

Do we know why? Do you see a configuration warning that the config change is ignored?

Reply:
As described in the commit message, "Setting a boolean Kconfig option default to "n" after having set it to "y" does not make the option disabled."

Reply:
What is a "higher priority Kconfig file"?

Reply:
> What is a "higher priority Kconfig file"?

Guess, it should be something like a .defconfig priority list: project-specific, board-specific, architecture-specific etc.

Reply:
No. Check this description: https://docs.zephyrproject.org/latest/build/kconfig/setting.html#multiple-symbol-definitions

> When a symbol is defined in multiple locations, each definition acts as an independent symbol that happens to share the same name. This means that properties are not appended to previous definitions. If the conditions for ANY definition result in the symbol resolving to y, the symbol will be y. It is therefore not possible to make the dependencies of a symbol more restrictive by defining it in multiple locations.

Reply:
Yes and if you look in the menuconfig you can see the dependencies of the different symbols named the same are OR'd together, not AND'd.

Reply:
unfortunate feature... probably have errors in other places ... learn something new every day ;-)